toolbox/fdc3-workbench/src/components/AppChannels.tsx

```diff
@@ -0,0 +1,361 @@
+import { ContextTemplates } from "./ContextTemplates";
+
+const useStyles = makeStyles((theme: Theme) =>
+	createStyles({
+		root: {
+			flexGrow: 1,
+		},
+		title: {
+			display: "flex",
+			alignItems: "center",
+			justifyContent: "space-between",
+		},
+		form: {
+			display: "flex",
+			flexWrap: "wrap",
+			alignItems: "center",
+			marginTop: theme.spacing(1),
+			"& > *": {
+				margin: theme.spacing(1),
+				marginLeft: "0px",
+			},
+		},
+        contextType: {
+			flexGrow: 1,
+			minWidth: "190px",
+		},
+        topMargin: {
+            marginTop: theme.spacing(2),
+        },
+		controls: {
+			"& > *:first-child": {
+				marginLeft: 0,
+			},
+			"& > *": {
+				marginRight: theme.spacing(1),
+			},
+			"& > *:last-child": {
+				marginRight: 0,
+			},
+			"& .MuiIconButton-sizeSmall": {
+				padding: "6px",
+			},
+		},
+		rightAlign: {
+			flexDirection: "row",
+			justifyContent: "flex-end",
+		},
+		spread: {
+			flexDirection: "row",
+			"& > *:first-child": {
+				paddingLeft: "0px",
+			},
+		},
+		textField: {
+			width: "100%",
+			"& input": {
+				height: "29px",
+				padding: "6px",
+			},
+		},
+		h4: {
+			fontSize: "22px",
+		},
+        h6: {
+            fontSize: "14px"
+        },
+		field: {
+			flexGrow: 1,
+			marginRight: theme.spacing(1),
+			minWidth: "190px",
+		},
+		border: {
+			height: "1px",
+			width: "100%",
+			backgroundColor: "#acb2c0",
+			marginTop: "24px",
+			marginBottom: "16px",
+		},
+        rightPadding: {
+            paddingRight: theme.spacing(.5),
+        },
+	})
+);
+
+interface ListenerOptionType {
+	title: string;
+	value: string;
+	type: string | undefined;
+}
+
+interface FilterOptionsState<T> {
+	inputValue: string;
+	getOptionLabel: (option: T) => string;
+}
+
+const listenerFilter = createFilterOptions<ListenerOptionType>();
+
+export const AppChannels = observer(({handleTabChange} : {handleTabChange:any}) => {
+    const classes = useStyles();
+    const [render, setRender] = useState<boolean>(false);
+    const [currentAppChannelId, setCurrentAppChannelId] = useState<string>("");
+    const contextListenersOptionsAll: ListenerOptionType[] = contextStore.contextsList.map(({ id, template }) => {
+        return {
+			title: id,
+			value: id,
+			type: template?.type
+		};
+	});
+	contextListenersOptionsAll.unshift({
+		title: "All",
+		value: "all",
+		type: "All"
+	});
+
+    const contextListenersOptions = Array.from(new Map(contextListenersOptionsAll.reverse().map((item) => [item["type"], item])).values()).reverse();
+
+
+    const handleGetorCreateChannel = (e: FormEvent | null = null) =>{
+        e?.preventDefault()
+        if (currentAppChannelId) {
+           let foundChannel = appChannelStore.appChannelsList.find((currentChannel)=>currentChannel.id === currentAppChannelId);
+            if (!foundChannel) {
+                appChannelStore.getOrCreateChannel(currentAppChannelId);
+            }
+           setCurrentAppChannelId("");
```

Comment:
this is set to always be an empty string because once the channel is created or gotten, the channelid string should be cleared. I have added that a value attribute the text field that modifies `currentAppChannelId`, so that when the channel is created the text field is cleared and the button disabled. This will come in a later commit. I will follow up when this pr is updated.